### PR TITLE
ddl: fix a bug that adding range partition meets error when value is negative

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

```
CREATE TABLE `tt5` (
  `c3` bigint(20) NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE ( `c3` ) (
  PARTITION p0 VALUES LESS THAN (-3),
  PARTITION p1 VALUES LESS THAN (-2)
);

ALTER TABLE `tt5` add partition ( partition p2 values less than (-1) );
(1054, u"Unknown column '' in 'partition function'")  # Succeed in MySQL, fail in TiDB.
```

and also this one:

```
ALTER TABLE `tt5` add partition ( partition p2 values less than (6 - 1) );
```

### What is changed and how it works?

In this line https://github.com/pingcap/tidb/compare/master...tiancaiamao:fix-add-range-partition?expand=1#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2L3395, we check the expression to be an integer.
However, `expr.GetType()` returns unspecified type and the error is throw out.

After removing that, I meet another error, here https://github.com/pingcap/tidb/compare/master...tiancaiamao:fix-add-range-partition?expand=1#diff-edfa54cbe2fdb418f0a77bd0ec4a56f2L2145

`checkAddPartitionValue` this function also asserts the partition less than value to be a simple integer,
so I replace it with another one.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @crazycs520 @zimulala @winkyao

**Comment by tiancaiamao** [comment]
PTAL @crazycs520 @zimulala @winkyao

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -3422,32 +3427,17 @@ func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec)
...
 		return nil, err
 	}
-	buf := new(bytes.Buffer)
 	for ith, def := range spec.PartDefinitions {
 		if err := def.Clause.Validate(part.Type, len(part.Columns)); err != nil {
 			return nil, errors.Trace(err)
 		}
 		// For RANGE partition only VALUES LESS THAN should be possible.
 		clause := def.Clause.(*ast.PartitionDefinitionClauseLessThan)
-		for _, expr := range clause.Exprs {
```
Comment (winkyao) [inline ddl/ddl_api.go]: Why just remove it? Could we refine the checking?

Comment (tiancaiamao) [reply ddl/ddl_api.go]: As you see, there are no test cases break and I'm adding new test cases, so it's fine to remove it.

**Pull request merged**